loopback-db-migrate.js

```diff
@@ -1,129 +1,118 @@
+
+const dateSinceFlag = process.argv.indexOf('--since');
+const dateSinceFilter = (dateSinceFlag > -1) ? process.argv[dateSinceFlag + 1] : '';
+
+const migrationsFolderFlag = process.argv.indexOf('--directory');
+const migrationsFolder = process.cwd() + (migrationsFolderFlag > -1 ? process.argv[migrationsFolderFlag + 1].replace(/\/?$/, '/') : '/server/migrations/');
+const dbMigrationsFolder = migrationsFolder + dbName;
+
+const appScriptFlag = process.argv.indexOf('--app-script');
+const appScript = appScriptFlag > -1 ? process.argv[appScriptFlag + 1] : '/server/server.js';
+const app = require(process.cwd() + appScript);
+
+app.on('booted', () => {
+  const datasource = app.dataSources[dbName];
+
+  if (!datasource) {
+    console.log('datasource \'' + dbName + '\' not found!');
+    process.exit(1);
   }
-});
-
-// make migration folders if they don't exist
-try {
-  fs.mkdirSync(migrationsFolder);
-} catch (e) {}
-try {
-  fs.mkdirSync(dbMigrationsFolder);
-} catch (e) {}
-
-function mapScriptObjName(scriptObj){
-  return scriptObj.name;
-}
-
-function findScriptsToRun(upOrDown, cb) {
-  var filters = {
-    where: {
-      name: { gte: dateSinceFilter+'' || '' }
-    },
-    order: (upOrDown === 'up' ) ? 'name ASC' : 'name DESC'
-  };
 
-  // get all local scripts and filter for only .js files
-  var localScriptNames = fs.readdirSync(dbMigrationsFolder).filter(function(fileName) {
-    return fileName.substring(fileName.length - 3, fileName.length) === '.js';
+  datasource.createModel('Migration', {
+    "name": {
+      "id": true,
+      "type": "String",
+      "required": true,
+      "length": 100
+    },
+    "db": {
+      "type": "String",
+      "length": 100,
+      "required": true
+    },
+    "runDtTm": {
+      "type": "Date",
+      "required": true
+    }
   });
 
-  // create table if not exists
-  datasource.autoupdate('Migration', function (err) {
-    if (err) {
-      console.log('Error retrieving migrations:');
-      console.log(err.stack);
-      process.exit(1);
-    }
+  // make migration folders if they don't exist
+  try {
+    fs.mkdirSync(migrationsFolder);
+  } catch (e) {
+  }
+
+  try {
+    fs.mkdirSync(dbMigrationsFolder);
+  } catch (e) {
+  }
 
-    // get all scripts that have been run from DB
-    datasource.models.Migration.find(filters, function (err, scriptsRun) {
+  function mapScriptObjName(scriptObj) {
+    return scriptObj.name;
+  }
+
+  function findScriptsToRun(upOrDown, cb) {
+    const filters = {
+      where: {
+        name: {gte: dateSinceFilter + '' || ''}
+      },
+      order: (upOrDown === 'up') ? 'name ASC' : 'name DESC'
+    };
+
+    // get all local scripts and filter for only .js files
+    const localScriptNames = fs.readdirSync(dbMigrationsFolder).filter(function (fileName) {
+      return fileName.substring(fileName.length - 3, fileName.length) === '.js';
+    });
+
+    // create table if not exists
+    datasource.autoupdate('Migration', function (err) {
```

Comment:
Do we think we should make this configurable? i.e. am-report-processor writing to a different migration table than that of the backend-api.

Reply:
Yeah, we can do that.

Reply:
This has now been done.